From 497fce4461611feb949bdb6f8cc7ea251552f02a Mon Sep 17 00:00:00 2001
From: Martin Jambor <mjambor@suse.cz>
Date: Wed, 10 Jul 2024 14:37:49 +0200
Subject: [PATCH] Fix GCC 14 initialization from incompatible pointer error
Upstream: project archived
References: boo#1225992

In file lib/ipmi_fru.c in function fru_area_print_multirec, local
variable mmh which is of type "struct fru_multirec_mgmt *" is
initialized with a type-cast to type "(struct fru_multirect_mgmt *)" -
note the t in multirect.  This causes a compile error by default with
GCC 14.  This patch fixes the typo to avoid the issue.

Signed-off-by: Martin Jambor <mjambor@suse.cz>
---
 lib/ipmi_fru.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/ipmi_fru.c b/lib/ipmi_fru.c
index 4d1dbbb..90b2a63 100644
--- a/lib/ipmi_fru.c
+++ b/lib/ipmi_fru.c
@@ -1553,7 +1553,7 @@ fru_area_print_multirec(struct ipmi_intf * intf, struct fru_info * fru,
 		case FRU_RECORD_TYPE_MANAGEMENT_ACCESS:
 			{
 				struct fru_multirec_mgmt *mmh =
-					(struct fru_multirect_mgmt *)
+					(struct fru_multirec_mgmt *)
 					&fru_data[sizeof(struct fru_multirec_header)];
 				size_t datalen = h->len - sizeof(*mmh);
 				struct {
-- 
2.45.2

